openfe_gromacs/protocols/gromacs_md/md_settings.py

```diff
@@ -0,0 +1,668 @@
+    Default 0.
+    """
+    nstvout: int = 0
+    """
+    Number of steps that elapse between writing velocities to the output
+    trajectory file (trr), the last velocities are always written unless 0,
+    which means velocities are not written into the trajectory file.
+    Default 0.
+    """
+    nstfout: int = 0
+    """
+    Number of steps that elapse between writing forces to the output trajectory
+    file (trr), the last forces are always written, unless 0, which means
+    forces are not written into the trajectory file.
+    Default 0.
+    """
+    nstlog: int = 1000
+    """
+    Number of steps that elapse between writing energies to the log file, the
+    last energies are always written. Default 1000.
+    """
+    nstcalcenergy: int = 100
+    """
+    Number of steps that elapse between calculating the energies, 0 is never.
+    This option is only relevant with dynamics. This option affects the
+    performance in parallel simulations, because calculating energies requires
+    global communication between all processes which can become a bottleneck at
+    high parallelization. Default 100.
+    """
+    nstenergy: int = 1000
+    """"
+    Number of steps that elapse between writing energies to energy file, the
+    last energies are always written, should be a multiple of nstcalcenergy.
+    Note that the exact sums and fluctuations over all MD steps modulo
+    nstcalcenergy are stored in the energy file, so gmx energy can report
+    exact energy averages and fluctuations also when nstenergy > 1
+    """
+    nstxout_compressed: int = 0
+    """
+    Number of steps that elapse between writing position coordinates using
+    lossy compression (xtc file), 0 for not writing compressed coordinates
+    output. Default 0.
+    """
+    compressed_x_precision: int = 1000
+    """
+    Precision with which to write to the compressed trajectory file.
+    Default 1000.
+    """
+    compressed_x_grps: str = ""
+    """
+    Group(s) to write to the compressed trajectory file, by default the whole
+    system is written (if nstxout-compressed > 0).
+    """
+    energygrps: str = ""
+    """
+    Group(s) for which to write to write short-ranged non-bonded potential
+    energies to the energy file (not supported on GPUs)
+    """
+
+    @validator(
+        "nstxout",
+        "nstvout",
+        "nstfout",
+        "nstlog",
+        "nstcalcenergy",
+        "nstenergy",
+        "nstxout_compressed",
+        "compressed_x_precision",
+    )
+    def must_be_positive_or_zero(cls, v):
+        if v < 0:
+            errmsg = (
+                "nstxout, nstvout, nstfout, nstlog, nstcalcenergy, "
+                "nstenergy, nstxout_compressed, and "
+                "compressed_x_precision must be zero or positive values,"
+                f" got {v}."
+            )
+            raise ValueError(errmsg)
+        return v
+
+
+class EMSimulationSettings(SimulationSettings):
```

Comment:
Ah this does exist - what do you think about making it so that dt isn't set forr this?